### PR TITLE
Increase abstractassert test coverage

### PR DESCRIPTION
**Pull request by sarajuhosova**
#### Check List:
* Fixes: NA
* Unit tests : YES
* Javadoc with a code example (on API only) : NA
* PR meets the [contributing guidelines](https://github.com/assertj/assertj-core/blob/main/CONTRIBUTING.md): NA

Following the [contributing guidelines](https://github.com/assertj/assertj-core/blob/main/CONTRIBUTING.md) will make it easier for us to review and accept your PR.

### CODE REVIEW

File: src/test/java/org/assertj/core/api/inputstream/InputStreamAssert_asString_with_charset_Test.java [hunk 1]
```diff
@@ -47,6 +54,23 @@ void should_convert_inputStream_to_a_proper_string_with_specific_encoding() {
...
+
+    // MOCK BEHAVIOUR
+    when(inputStream.read(any(), anyInt(), anyInt())).thenThrow(new IOException());
+
+    // ASSERT THROWS
+    assertThatThrownBy(() -> {
+      // WHEN/THEN
+      then(inputStream).asString(TURKISH_CHARSET)
+        .isEqualTo(real);
+    }).isInstanceOf(InputStreamsException.class);
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/inputstream/InputStreamAssert_asString_with_charset_Test.java:71]: Minor: restructure the test from the ASSERT THROWS comment as:
```java
   // WHEN/THEN
   thenThrownBy(() -> assertThat(inputStream).asString(TURKISH_CHARSET).isEqualTo(real)).isInstanceOf(InputStreamsException.class);
``` 
an alternative would be:
```java
   // WHEN
   ThrowingCallable assertionCall = () -> assertThat(inputStream).asString(TURKISH_CHARSET).isEqualTo(real);
   // WHEN/THEN
   thenThrownBy(assertionCall).isInstanceOf(InputStreamsException.class);
```
pick the one that is most readable to you ;-)

File: src/test/java/org/assertj/core/api/optional/OptionalAssert_contains_usingDefaultComparator_Test.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * Tests for <code>{@link OptionalAssert#usingDefaultComparator()}</code>.
+ *
+ * @author Sára Juhošová
+ */
+class OptionalAssert_contains_usingDefaultComparator_Test {
+
+  private final Comparator<String> STRING_COMPARATOR = Comparator.comparing(String::toLowerCase);
+
+  @Test
+  void should_succeed_if_default_equal_content() {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/optional/OptionalAssert_contains_usingDefaultComparator_Test.java:34]: add // GIVEN

File: src/test/java/org/assertj/core/api/optional/OptionalAssert_contains_usingDefaultComparator_Test.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    OptionalAssert<String> optionalAssert = assertThat(Optional.of("hello"));
+    String expected = "hello";
+
+    // set to different strategy
+    optionalAssert.usingValueComparator(STRING_COMPARATOR).contains(expected);
+
+    // go back to default strategy
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/optional/OptionalAssert_contains_usingDefaultComparator_Test.java:41]: add // WHEN/THEN

File: src/test/java/org/assertj/core/api/optional/OptionalAssert_contains_usingDefaultComparator_Test.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    OptionalAssert<String> optionalAssert = assertThat(Optional.of("hello"));
+    String expected = "HellO";
+
+    // set to different strategy
+    optionalAssert.usingValueComparator(STRING_COMPARATOR).contains(expected);
+
+    assertThatExceptionOfType(AssertionError.class).isThrownBy(() ->
+        // go back to default strategy
+        optionalAssert.usingDefaultValueComparator().contains(expected)
+      );
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/optional/OptionalAssert_contains_usingDefaultComparator_Test.java:56]: use the simpler `expectAssertionError` from `AssertionsUtil` over `assertThatExceptionOfType(AssertionError.class)`

**Review by joel-costigliola — CHANGES_REQUESTED** [review]
thanks a lot @sarajuhosova for this great PR.
I have a few comments that are mainly stylistic and for simplifying the code a bit more.

**Comment by sarajuhosova** [comment]
Thanks, @joel-costigliola, for the feedback, I'll fix the style issues asap. I was also planning to add tests for the more complex classes that I've left out for now, should I add it to this PR or make a new on to make reviewing a bit easier?

**Comment by joel-costigliola** [comment]
thanks for the quick reply, I would prefer another PR so that we can finish this one and it's easier to review shorter PRs.

**Comment by sarajuhosova** [comment]
I've fixed the commented things, hopefully this is what you meant!

**Comment by joel-costigliola** [comment]
Integrated thanks @sarajuhosova!
I noticed when integrating some unused imports or `*` imports and I tool the liberty to reformat other tests not initially in the PR and squashed all the commits (keeping you as the commit author obviously)

**Pull request closed without merging**